cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/questionnaire/generate/ItemGenerator.java

```diff
@@ -36,55 +41,72 @@ public class ItemGenerator {
 
     protected final Repository repository;
     protected final CqlEngine engine;
-    protected final IElementProcessor elementProcessor;
     protected final ExpressionProcessor expressionProcessor;
     protected final ExtensionProcessor extensionProcessor;
+    protected final ElementHasCaseFeature elementHasCaseFeature;
+    protected final ItemTypeIsChoice itemTypeIsChoice;
 
     public ItemGenerator(Repository repository) {
-        this(repository, IElementProcessor.createProcessor(repository));
-    }
-
-    public ItemGenerator(Repository repository, IElementProcessor elementProcessor) {
         this.repository = repository;
         engine = Engines.forRepository(this.repository);
-        this.elementProcessor = elementProcessor;
         expressionProcessor = new ExpressionProcessor();
         extensionProcessor = new ExtensionProcessor();
+        elementHasCaseFeature = new ElementHasCaseFeature();
+        itemTypeIsChoice = new ItemTypeIsChoice(repository);
     }
 
     public <T extends IBaseExtension<?, ?>> Pair<IBaseBackboneElement, List<T>> generate(GenerateRequest request) {
```

Comment:
Please annotate any methods that could return null with jakarta.annotation.Nullable